cns/restserver/api.go

```diff
@@ -1029,7 +1030,16 @@ func (service *HTTPRestService) unpublishNetworkContainer(w http.ResponseWriter,
 
 	ctx := r.Context()
 
-	if !service.isNetworkJoined(req.NetworkID) {
+	var unpublishBody nmagent.DeleteContainerRequest
+	if req.DeleteNetworkContainerRequestBody != nil {
+		err = json.Unmarshal(req.DeleteNetworkContainerRequestBody, &unpublishBody)
+		if err != nil {
+			http.Error(w, fmt.Sprintf("could not unmarshal delete network container body: %v", err), http.StatusBadRequest)
+			return
+		}
+	}
+
```

Comment:
can you add more comments indicating why we need to join for AZR NCs?

Reply:
added a comment below